From 87d1c61eec8a8e3de298c6384714ddddd3f04949 Mon Sep 17 00:00:00 2001
From: Hoang Quoc Trung <quoctrunghoang1998@gmail.com>
Date: Tue, 1 Mar 2022 17:44:53 +0100
Subject: Reduce test duration for google mock CA (#37597)

* Reduce sleep delay for mock CA server

Signed-off-by: Trung <trung.hoang@pricehubble.com>

* Reduce sleep duration for google-cas mock

Signed-off-by: Trung <trung.hoang@pricehubble.com>

* Remove sleep safeguard

Signed-off-by: Trung <trung.hoang@pricehubble.com>
---
 .../nodeagent/caclient/providers/google-cas/mock/ca_mock.go  | 2 +-
 .../pkg/nodeagent/caclient/providers/google/mock/ca_mock.go  | 5 +----
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/security/pkg/nodeagent/caclient/providers/google-cas/mock/ca_mock.go b/security/pkg/nodeagent/caclient/providers/google-cas/mock/ca_mock.go
index 6bfad90204..cd4a18d966 100644
--- a/security/pkg/nodeagent/caclient/providers/google-cas/mock/ca_mock.go
+++ b/security/pkg/nodeagent/caclient/providers/google-cas/mock/ca_mock.go
@@ -156,7 +156,7 @@ func CreateServer(service *CASService) (*CASServer, *bufconn.Listener, error) {
 	}()
 
 	select {
-	case <-time.After(1 * time.Second):
+	case <-time.After(100 * time.Millisecond):
 		err = nil
 	case err = <-serveErr:
 	}
diff --git a/security/pkg/nodeagent/caclient/providers/google/mock/ca_mock.go b/security/pkg/nodeagent/caclient/providers/google/mock/ca_mock.go
index 9f8ab4a4c7..94e67b6ccf 100644
--- a/security/pkg/nodeagent/caclient/providers/google/mock/ca_mock.go
+++ b/security/pkg/nodeagent/caclient/providers/google/mock/ca_mock.go
@@ -18,7 +18,6 @@
 	"context"
 	"fmt"
 	"net"
-	"time"
 
 	"google.golang.org/grpc"
 
@@ -62,15 +61,13 @@ func CreateServer(addr string, service *CAService) (*CAServer, error) {
 	s.Address = lis.Addr().String()
 
 	var serveErr error
+	gcapb.RegisterMeshCertificateServiceServer(s.Server, service)
 	go func() {
-		gcapb.RegisterMeshCertificateServiceServer(s.Server, service)
 		if err := s.Server.Serve(lis); err != nil {
 			serveErr = err
 		}
 	}()
 
-	// The goroutine starting the server may not be ready, results in flakiness.
-	time.Sleep(1 * time.Second)
 	if serveErr != nil {
 		return nil, err
 	}
-- 
2.35.3

